### PR TITLE
WIP - ticket:5490 Enable new instantiation and API by default

### PR DESCRIPTION
**Pull request by adeas31**

### CODE REVIEW

File: OMEdit/OMEdit/OMEditGUI/Options/OptionsDialog.cpp [hunk 1]
```diff
@@ -3661,6 +3669,10 @@ SimulationPage::SimulationPage(OptionsDialog *pOptionsDialog)
   mpIgnoreCommandLineOptionsAnnotationCheckBox = new QCheckBox(tr("Ignore __OpenModelica_commandLineOptions annotation"));
   // ignore simulation flags annotation checkbox
   mpIgnoreSimulationFlagsAnnotationCheckBox = new QCheckBox(tr("Ignore __OpenModelica_simulationFlags annotation"));
+  //! @todo Remove this once we enable this bydefault in OMC.
+  /* Enable new instantiation use in OMC API */
+  mpEnableNewInstantiationAPICheckBox = new QCheckBox(tr("Enable experimental new instantiation use in the OMC API"));
```
Comment (casella) [inline OMEdit/OMEdit/OMEditGUI/Options/OptionsDialog.cpp]: I would suggest "Enable new front end use in the OMC API (faster GUI response)". We always referred to "new front end" when marketing this thing, I'm afraid "new instantiation" would be confusing. Also, this is no longer supposed to be experimental, once beta-tested, the aim of 1.14.0 is to have this thing production-grade

File: OMEdit/OMEdit/OMEditGUI/Simulation/TranslationFlagsWidget.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ TranslationFlagsWidget::TranslationFlagsWidget(QWidget *pParent)
   mpEvaluateAllParametersCheckBox = new QCheckBox(tr("Evaluate all parameters (faster simulation, cannot change them at runtime)"));
   mpNLSanalyticJacobianCheckBox = new QCheckBox(tr("Enable analytical jacobian for non-linear strong components"));
   mpParmodautoCheckBox = new QCheckBox(tr("Enable parallelization of independent systems of equations (Experimental)"));
-  mpNewInstantiationCheckBox = new QCheckBox(tr("Enable experimental new instantiation phase"));
+  mpOldInstantiationCheckBox = new QCheckBox(tr("Enable old instantiation phase"));
```
Comment (casella) [inline OMEdit/OMEdit/OMEditGUI/Simulation/TranslationFlagsWidget.cpp]: I would use "Old front end" instead, see previous comment

**Comment by adeas31** [comment]
Popup for old frontend if the new one fails,

Screen

**Comment by adeas31** [comment]
Settings for old frontend. In case you want to change the remembered decision. By default new frontend is enabled and user will get a popup for old frontend in case of the failure.

Screen1

**Comment by casella** [comment]
Looks good. It may still be a bit confusing for beginners, who have no clue what the frontend is, never mind the old one and the new one.

Maybe we could write "The code generation process failed, see the Messages Browser for detailed diagnostic messages." and "but it could also be due to some issues with the new OpenModelica compiler frontend".

Also, I would suggest to move the "Keep using new frontend" button to the left and make it the default choice. As you correctly wrote, in most cases the errors are due to issues with the Modelica source code, so the default behaviour should be to keep the new front end and find out the bug by looking at the error messages.

**Comment by adeas31** [comment]
Should I also make  "Keep using new frontend" a default button?

**Comment by adeas31** [comment]
image

**Comment by casella** [comment]
> 
> 
> Should I also make "Keep using new frontend" a default button?

Yes. If you don't know better, that's the right choice 99% of the times.

**Comment by casella** [comment]
Unless you have models with arrays of records, that is... (@perost)

**Comment by casella** [comment]
@adeas31 I installed the latest nightly today. I got "Enable old front end for code generation" selected in the Tools|Options|Simulation windows, which should not be the case, we agreed that the default should be to use the new front-end.

Also, I couldn't get the the popup windows to switch to the old front end, when trying to run models that would make the front end fail because of a syntax error.

Last, but not least, it's not clear to me if the new API is actually used. I would say so, given the speed with which the MultiBody examples are handled, but I'm not 100% sure. Can you please confirm?

Thanks!

**Comment by casella** [comment]
> Last, but not least, it's not clear to me if the new API is actually used. I would say so, given the speed with which the MultiBody examples are handled, but I'm not 100% sure. Can you please confirm?

Forget about this, I just needed to scroll down a bit...

**Comment by casella** [comment]
@adeas31, I tried to build a simple MultiBody model, where I set the initial conditions in a way that missed 'each', which the NF doesn't like. I got the popup, tried the old front-end, and it worked like a charm.

I'm not sure why I didn't get the popup previously, is there some heuristic in place to rule out errors which are obviously not the NF's fault?

**Comment by adeas31** [comment]
> @adeas31 I installed the latest nightly today. I got "Enable old front end for code generation" selected in the Tools|Options|Simulation windows, which should not be the case, we agreed that the default should be to use the new front-end.

I have used the same flag as we have for the new frontend but just reversed its value. Maybe be in your settings you have new frontend as false previously. I re-checked that new frontend is enabled by default.

> I'm not sure why I didn't get the popup previously, is there some heuristic in place to rule out errors which are obviously not the NF's fault?

No. It just checks if `translateModel` is successful or not.

**Comment by casella** [comment]
OK, in fact I don't get the popup if there are syntax errors, but that's ok, the NF shouldn't have problems with that.

I will close the ticket, the new configuration seems to be working fine.

**Pull request merged**